bin/update-gh-pages-documentation-site

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+#
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2022 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See https://swift.org/LICENSE.txt for license information
+# See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+#
+# Updates the GitHub Pages documentation site thats published from the 'docs' 
+# subdirectory in the 'gh-pages' branch of this repository.
+#
+# This script should be run by someone with commit access to the 'gh-pages' branch
+# at a regular frequency so that the documentation content on the GitHub Pages site
+# is up-to-date with the content in this repo.
+#
+
+set -eu
+
+# A `realpath` alternative using the default C implementation.
+filepath() {
```

Comment:
This isn't great. Fails to resolve symlinks properly AFAIK. When I get back home I can send you a version of this that is a bit more robust.

Reply:
This is what we use everywhere in swift project. Such as the following 👇

https://github.com/apple/swift-docc/blob/b4d2c4208d7ae3da2ec39d1a97d180d28ee73b82/bin/test#L15-L17

https://github.com/apple/swift-docc-plugin/blob/859caac534e94ace18b894ccd9ed301ae4aeda84/bin/test#L15-L17

https://github.com/apple/swift-markdown/blob/e50693584310a9190071c96c839485b6f2376832/bin/update-gh-pages-documentation-site#L22-L24

https://github.com/apple/swift-markdown/blob/e50693584310a9190071c96c839485b6f2376832/bin/test#L15-L17


Maybe a follow-up PR to update them all?

Reply:
Sounds good, hopefully I will get to it at some point. For future reference this is what I have been using:
```bash
# Intended to be used in a subshell.
filepath() {
    local _target_file
    _target_file="$1"

    cd "$(dirname "${_target_file}")"
    _target_file=$(basename "${_target_file}")

    # Iterate down a (possible) chain of symlinks
    while [[ -L "${_target_file}" ]]; do
        _target_file=$(readlink "${_target_file}")
        cd "$(dirname "${_target_file}")"
        _target_file=$(basename "${_target_file}")
    done

    # Compute the canonicalized name by finding the physical path for
    # the directory we're in and appending the target file.
    echo "$(pwd -P)/${_target_file}"
}
```